PolyEngine/Engine/Src/KeyBindings.hpp

```diff
@@ -182,6 +182,187 @@ namespace Poly
+		NUM_SEPARATOR          = UNDEFINED, //?
+		NUM_SUBTRACT           = 0x4e,
+		NUM_DECIMAL            = 0x41,
+		NUM_DIVIDE             = 0x4b, //where is NUM_EQUALS or NUM_ENTER ?
+		F1                     = 0x7a,
+		F2                     = 0x78,
+		F3                     = 0x63,
+		F4                     = 0x76,
+		F5                     = 0x60,
+		F6                     = 0x61,
+		F7                     = 0x62,
+		F8                     = 0x64,
+		F9                     = 0x65,
+		F10                    = 0x6d,
+		F11                    = 0x67,
+		F12                    = 0x6f,
+		F13                    = 0x69,
+		F14                    = 0x6b,
+		F15                    = 0x71,
+		F16                    = 0x6a,
+		F17                    = 0x40,
+		F18                    = 0x4f,
+		F19                    = 0x50,
+		F20                    = 0x5a,
+		F21                    = UNDEFINED, // No more F keys on mac keyboard
+		F22                    = UNDEFINED,
+		F23                    = UNDEFINED,
+		F24                    = UNDEFINED,
+		NUM_LOCK               = UNDEFINED, // Nope
+		SCROLL_LOCK            = UNDEFINED, // Nope
+		LSHIFT                 = 0x38,
+		RSHIFT                 = 0x3c,
+		LCTRL                  = 0x3b,
+		RCTRL                  = 0x3e,
+		LMENU                  = UNDEFINED, // Nope
+		RMENU                  = UNDEFINED,
+		BROWSER_BACK           = UNDEFINED,
+		BROWSER_FORWARD        = UNDEFINED,
+		BROWSER_REFRESH        = UNDEFINED,
+		BROWSER_STOP           = UNDEFINED, // Nope
+		BROWSER_SEARCH         = UNDEFINED,
+		BROWSER_FAVORITES      = UNDEFINED,
+		BROWSER_HOME           = UNDEFINED,
+		VOLUME_MUTE            = 0x4a,
+		VOLUME_DOWN            = 0x49,
+		VOLUME_UP              = 0x48,
+		MEDIA_NEXT             = UNDEFINED,
+		MEDIA_PREV             = UNDEFINED,
+		MEDIA_STOP             = UNDEFINED,
+		MEDIA_PLAY_PAUSE       = UNDEFINED,
+		LAUNCH_MAIL            = UNDEFINED, //:(
+		LAUNCH_MEDIA_SELECT    = UNDEFINED,
+		LAUNCH_APP1            = UNDEFINED,
+		LAUNCH_AP2             = UNDEFINED,
+		SEMICOLON_COLON        = 0x29,
+		PLUS                   = 0x18, // WTF, is this equals key?
+		COMMA                  = 0x2b,
+		MINUS                  = 0x1b,
+		PERIOD                 = 0x2f,
+		SLASH_QUESTION         = 0x2c,
+		TILDE                  = 0x32,
+		OPEN_SQUARE_CURLY      = 0x21,
+		BACKSLASH_VERTICAL_BAR = 0x2a,
+		CLOSE_SQUARE_CURLY     = 0x1e,
+		QUOTE_SINGLE_DOUBLE    = 0x27,
+		OEM_8                  = UNDEFINED, //?
+		OEM_102                = UNDEFINED,
+		IME_PROCESS            = UNDEFINED,
+		PACKET                 = UNDEFINED,
+		ATTN                   = UNDEFINED,
+		CRSEL                  = UNDEFINED,
+		EXSEL                  = UNDEFINED,
+		EREOF                  = UNDEFINED,
+		PLAY                   = UNDEFINED,
+		ZOOM                   = UNDEFINED,
+		PA1                    = UNDEFINED,
+		OEM_CLEAR              = UNDEFINED,
+		_COUNT                 = 0xFF
+	};
+}
```

Comment:
These could use some cleaning up on all platforms. Lots of keys will never be used or are very unlikely to. Some are weirdly named. And there's an issue of Ctrl/Shift/Alt resolving to undefined on Linux (we should probably handle this higher up and set SHIFT whenever any of LSHIFT and RSHIFT are pressed and so on).